src/client/js/Panels/MetaEditor/MetaEditorControl.js

```diff
@@ -1174,46 +1174,71 @@ define(['js/logger',
     };
 
     MetaEditorControl.prototype._createPointerRelationship = function (sourceID, targetID, isSet) {
-        var sourceNode = this._client.getNode(sourceID),
-            targetNode = this._client.getNode(targetID),
+        var client = this._client,
+            sourceNode = client.getNode(sourceID),
+            targetNode = client.getNode(targetID),
+            metaNodes = client.getAllMetaNodes(true),
             pointerMetaDescriptor,
+            existingNames,
             existingPointerNames,
-            notAllowedPointerNames,
-            self = this;
+            notAllowedPointerNames;
+
+        function getExistingNames() {
```

Comment:
how would this function deal with the situation when typeA has two instances typeAone and typeAtwo that both introduces new names?
why the existing names do not cover all existing names throughout the Meta (or I read it wrong?)?